### ISSUE TITLE
Add an easy way to check if Capistrano is in dry run mode

### ISSUE THREAD

**Issue by mattbrictson** [capistrano/capistrano#1564]
Perhaps we could add `dry_run?` to the DSL so that e.g. plugin authors can know not to make actual API requests, etc.

One use case from @phallstrom (see #1462):

> My case is that I maintain slackistrano. A gem to post deploy notifications to Slack. My problem is if users do a dry run those notifications are still posted. So I need a way to know if we're in a dry run so I can do what SSH does and simply print "would have posted to a slack channel", etc.

**Comment by mattbrictson** [comment]
This should be a fairly simple enhancement if someone wants to take a crack at a PR. I'm thinking just a `Configuration#dry_run?` method that checks if the backend is a `Printer`, and a corresponding method on `DSL::Env` that delegates to it.

Any takers? /cc @phallstrom @rainchen

**Comment by dgtized** [comment]
Including the following in the Capfile addresses this:

```ruby
module Capistrano
  def self.dry_run?
    Configuration.env.fetch(:sshkit_backend) == SSHKit::Backend::Printer
  end
end
```

**Comment by mattbrictson** [comment]
@dgtized Looks good! Would you be willing to submit a PR with a test?

**Comment by kylerippey** [comment]
I believe this issue has been resolved by #1640.
